File: app/src/main/java/com/dragonpav/dengine/BoundingBox.java
package com.dragonpav.dengine;

public class BoundingBox {
    public Values.Vector3 min;
    public Values.Vector3 max;
    public boolean contains(Values.Vector3 vec) {
        return (vec.x >= min.x && vec.x <= max.x) &&
                (vec.y >= min.y && vec.y <= max.y) &&
                (vec.z >= min.z && vec.z <= max.z);
    };
    public Values.Vector3 getSizes() {
        float sizeX = max.x - min.x;
        float sizeY = max.y - min.y;
        float sizeZ = max.z - min.z;
        return new Values.Vector3(sizeX, sizeY, sizeZ);
    }
    public Values.Vector3 getCenter() {
        return new Values.Vector3((min.x + max.x) / 2.0f,
                (min.y + max.y) / 2.0f,
                (min.z + max.z) / 2.0f);
    }
    public BoundingBox() {
        min = new Values.Vector3();
        max = new Values.Vector3();
    };
    public BoundingBox(Values.Vector3 min, Values.Vector3 max) {
        this.min = min;
        this.max = max;
    }
}

File: src/BoundingBox.cpp
#include "BoundingBox.h"
BoundingBox::BoundingBox(Vector3 min, Vector3 max) {
	this->min = min;
	this->max = max;
}
void BoundingBox::setMin(Vector3 minV) {
	min = minV;
}
void BoundingBox::setMax(Vector3 maxV) {
	max = maxV;
}
Vector3& BoundingBox::getMin() {
	return min;
}
Vector3& BoundingBox::getMax() {
	return max;
}
bool BoundingBox::contains(Vector3 vec) {
	return (vec.getX() >= min.getX() && vec.getX() <= max.getX()) &&
		(vec.getY() >= min.getY() && vec.getY() <= max.getY()) &&
		(vec.getZ() >= min.getZ() && vec.getZ() <= max.getZ());
}
Vector3 BoundingBox::getSizes() {
	float sizeX = max.getX() - min.getX();
	float sizeY = max.getY() - min.getY();
	float sizeZ = max.getZ() - min.getZ();
	return Vector3(sizeX, sizeY, sizeZ);
}
Vector3 BoundingBox::getCenter() {
	return Vector3((min.getX() + max.getX()) / 2.0f,
		(min.getY() + max.getY()) / 2.0f,
		(min.getZ() + max.getZ()) / 2.0f);
}